62.CAD/main.cpp

```diff
@@ -234,52 +407,866 @@ class CPolyline
+	// This gives us a quartic equation:
+	// At^4 + Bt^3 + Ct^2 + Dt + E = 0
+	//
+	// The roots for t then become our intersections points between both curves.
+	//
+	// A Desmos page including math for this as well as some of the graphs it generates is available here:
+	// https://www.desmos.com/calculator/mjwqvnvyb8?lang=pt-BR
+
+	double p0x = p[0].x, p1x = p[1].x, p2x = p[2].x,
+		p0y = p[0].y, p1y = p[1].y, p2y = p[2].y;
+
+	// Getting the values for the implicitization of the curve
+	// TODO: Do this with quadratic coefficients instead (A, B, C)
+	double t0 = (4 * p0y * p1y) - (4 * p0y * p2y) - (4 * (p1y * p1y)) + (4 * p1y * p2y) - ((p0y * p0y)) + (2 * p0y * p2y) - ((p2y * p2y));
+	double t1 = -(4 * p0x * p1y) + (4 * p0x * p2y) - (4 * p1x * p0y) + (8 * p1x * p1y) - (4 * p1x * p2y) + (4 * p2x * p0y) - (4 * p2x * p1y) + (2 * p0x * p0y) - (2 * p0x * p2y) - (2 * p2x * p0y) + (2 * p2x * p2y);
+	double t2 = (4 * p0x * p1x) - (4 * p0x * p2x) - (4 * (p1x * p1x)) + (4 * p1x * p2x) - ((p0x * p0x)) + (2 * p0x * p2x) - ((p2x * p2x));
+	double t3 = (4 * p0x * (p1y * p1y)) - (4 * p0x * p1y * p2y) - (4 * p1x * p0y * p1y) + (8 * p1x * p0y * p2y) - (4 * p1x * p1y * p2y) - (4 * p2x * p0y * p1y) + (4 * p2x * (p1y * p1y)) - (2 * p0x * p0y * p2y) + (2 * p0x * (p2y * p2y)) + (2 * p2x * (p0y * p0y)) - (2 * p2x * p0y * p2y);
+	double t4 = -(4 * p0x * p1x * p1y) - (4 * p0x * p1x * p2y) + (8 * p0x * p2x * p1y) + (4 * (p1x * p1x) * p0y) + (4 * (p1x * p1x) * p2y) - (4 * p1x * p2x * p0y) - (4 * p1x * p2x * p1y) + (2 * (p0x * p0x) * p2y) - (2 * p0x * p2x * p0y) - (2 * p0x * p2x * p2y) + (2 * (p2x * p2x) * p0y);
+	double t5 = (4 * p0x * p1x * p1y * p2y) - (4 * (p1x * p1x) * p0y * p2y) + (4 * p1x * p2x * p0y * p1y) - ((p0x * p0x) * (p2y * p2y)) + (2 * p0x * p2x * p0y * p2y) - ((p2x * p2x) * (p0y * p0y)) - (4 * p0x * p2x * (p1y * p1y));
+
+	// "Slam" the other curve onto it
+
+	float64_t2 A = second.p[0] - 2.0 * second.p[1] + second.p[2];
+	float64_t2 B = 2.0 * (second.p[1] - second.p[0]);
+	float64_t2 C = second.p[0];
+
+	// Getting the quartic params
+	double a = ((A.x * A.x) * t0) + (A.x * A.y * t1) + (A.y * t2);
+	double b = (2 * A.x * B.x * t0) + (A.x * B.y * t1) + (B.x * A.y * t1) + (2 * A.y * B.y * t2);
+	double c = (2 * A.x * C.x * t0) + (A.x * C.y * t1) + (A.x * t3) + ((B.x * B.x) * t0) + (B.x * B.y * t1) + (C.x * A.y * t1) + (2 * A.y * C.y * t2) + (A.y * t4) + ((B.y * B.y) * t2);
+	double d = (2 * B.x * C.x * t0) + (B.x * C.y * t1) + (B.x * t3) + (C.x * B.y * t1) + (2 * B.y * C.y * t2) + (B.y * t4);
+	double e = ((C.x * C.x) * t0) + (C.x * C.y * t1) + (C.x * t3) + ((C.y * C.y) * t2) + (C.y * t4) + (t5);
+
+	return hatchutils::solveQuarticRoots(a, b, c, d, e, 0.0, 1.0);
+}
+
+double Hatch::QuadraticBezier::intersectOrtho(double coordinate, int major) const
+{
+	// https://pomax.github.io/bezierinfo/#intersections
+	double points[3];
+	for (uint32_t i = 0; i < 3; i++)
+		points[i] = major ? p[i].y : p[i].x;
+
+	for (uint32_t i = 0; i < 3; i++)
```

Comment:
when you have the bezier in ABC form then you only need to subtract coordinate from the c value